napari/layers/labels/_labels_mouse_bindings.py

```diff
@@ -26,57 +21,26 @@ def draw(layer, event):
     pixels will be changed to background and this tool functions like an
     eraser
     """
-    ndisplay = len(layer._dims_displayed)
     coordinates = mouse_event_to_labels_coordinate(layer, event)
 
     # on press
-    if layer._mode == Mode.ERASE:
-        new_label = layer._background_label
-    else:
-        new_label = layer.selected_label
-
-    if coordinates is not None:
-        if layer._mode in [Mode.PAINT, Mode.ERASE]:
-            layer.paint(coordinates, new_label)
-        elif layer._mode == Mode.FILL:
-            layer.fill(coordinates, new_label)
-    else:  # still add an item to undo queue
-        # when dragging, if we start a drag outside the layer, we will
-        # incorrectly append to the previous history item. We create a
-        # dummy history item to prevent this.
-        dummy_indices = (np.zeros(shape=0, dtype=int),) * layer.data.ndim
-        layer._undo_history.append([(dummy_indices, [], [])])
-
-    last_cursor_coord = coordinates
-    yield
+    with layer.block_history():
+        if layer._mode == Mode.ERASE:
+            new_label = layer._background_label
+        else:
+            new_label = layer.selected_label
```

Comment:
These lines belong outside the `with` block?

But otherwise, man, this looks so much cleaner! 😍